src/impl/apple/keychain_helper.cpp

```diff
@@ -50,66 +51,91 @@ CFPtr<CFStringRef> convert_string(const std::string& string)
     return result;
 }
 
-CFPtr<CFMutableDictionaryRef> build_search_dictionary(const std::string& account, const std::string& service,
+CFPtr<CFMutableDictionaryRef> build_search_dictionary(const CFStringPtr& account,
```

Comment:
Taking smart pointers by const reference is a little weird. Typically you'd take a raw pointer to indicate that you're merely borrowing a reference, rather than doing anything with the ownership of the object. For that reason I'd suggest just using `CFStringRef` for the string arguments.